tools/flavor-convert/main.go

```diff
@@ -0,0 +1,440 @@
+				continue
+			}
+
+			var newFlavorPcrs []types.PCRS
+			newFlavorPcrs = make([]types.PCRS, len(pcrsmap))
+
+			for bank, pcrMap := range flavor.Flavor.Pcrs {
+				index := 0
+				for mapIndex, templateBank := range pcrsmap {
+					pcrIndex := types.PcrIndex(mapIndex)
+
+					if types.SHAAlgorithm(bank) != types.SHAAlgorithm(templateBank) {
+						break
+					}
+					if expectedPcrEx, ok := pcrMap[pcrIndex.String()]; ok {
+						newFlavorPcrs[index].PCR.Index = mapIndex
+						newFlavorPcrs[index].PCR.Bank = bank
+						newFlavorPcrs[index].Measurement = expectedPcrEx.Value
+						newFlavorPcrs[index].PCRMatches = rules[index].PcrMatches
+
+						var newTpmEvents []types.EventLogCriteria
+						if expectedPcrEx.Event != nil && !reflect.ValueOf(rules[index].EventlogEquals).IsZero() {
+							newFlavorPcrs[index].EventlogEqual = new(types.EventLogEqual)
+							if rules[index].EventlogEquals.ExcludingTags != nil {
+								newFlavorPcrs[index].EventlogEqual.ExcludeTags = rules[index].EventlogEquals.ExcludingTags
+							}
+
+							newTpmEvents = make([]types.EventLogCriteria, len(expectedPcrEx.Event))
+							for eventIndex, oldEvents := range expectedPcrEx.Event {
+								newTpmEvents[eventIndex].TypeName = oldEvents.Label
+								newTpmEvents[eventIndex].Tags = append(newTpmEvents[eventIndex].Tags, oldEvents.Label)
+								newTpmEvents[eventIndex].Measurement = oldEvents.Value
+								newTpmEvents[eventIndex].TypeID = eventIDList[oldEvents.Label]
+							}
+							newFlavorPcrs[index].EventlogEqual.Events = newTpmEvents
+							newTpmEvents = nil
+						}
+
+						if expectedPcrEx.Event != nil && !reflect.ValueOf(rules[index].EventlogIncludes).IsZero() {
+							newTpmEvents = make([]types.EventLogCriteria, len(expectedPcrEx.Event))
+							for eventIndex, oldEvents := range expectedPcrEx.Event {
+								newTpmEvents[eventIndex].TypeName = oldEvents.Label
+								newTpmEvents[eventIndex].Tags = append(newTpmEvents[eventIndex].Tags, oldEvents.Label)
+								newTpmEvents[eventIndex].Measurement = oldEvents.Value
+								newTpmEvents[eventIndex].TypeID = eventIDList[oldEvents.Label]
+							}
+							newFlavorPcrs[index].EventlogIncludes = newTpmEvents
+							newTpmEvents = nil
+						}
+					}
+					index++
+				}
+			}
+			flavor.Flavor.PcrLogs = newFlavorPcrs
+		}
+		oldFlavorPart.SignedFlavor[flavorIndex].Flavor.Pcrs = nil
+		oldFlavorPart.SignedFlavor[flavorIndex].Flavor.PcrLogs = flavor.Flavor.PcrLogs
+	}
+
+	//getting the final data
+	finalFlavorPart, err := json.Marshal(oldFlavorPart.SignedFlavor)
+	if err != nil {
+		fmt.Println("Error in marshaling the final flavor part file")
+		os.Exit(1)
+	}
+
+	//Printing the final flavor part file in console
+	fmt.Println("New flavor part json:\n", string(finalFlavorPart))
+
+	//writing the new flavor part into the local file
+	data := []byte(finalFlavorPart)
+	err = ioutil.WriteFile("/opt/newflavorpart.json", data, 0644)
```

Comment:
better if we take this as an argument also the signing keys

Reply:
Here the output flavor part file is only a flavor collection not the signed flavor. So, we did not use any signing keys in flavor convert tool. Let us know your comments on this.